system-addon/content-src/components/StartupOverlay/StartupOverlay.jsx

```diff
@@ -0,0 +1,86 @@
+import {FormattedMessage, injectIntl} from "react-intl";
+import {actionCreators as ac} from "common/Actions.jsm";
+import {connect} from "react-redux";
+import React from "react";
+
+export class _StartupOverlay extends React.PureComponent {
+  constructor(props) {
+    super(props);
+    this.onInputChange = this.onInputChange.bind(this);
+    this.onSubmit = this.onSubmit.bind(this);
+    this.clickSkip = this.clickSkip.bind(this);
+    this.initScene = this.initScene.bind(this);
+
+    this.state = {emailInput: ""};
+  }
+
+  initScene() {
+    // Timeout to allow the scene to render once before attaching the attribute
+    // to trigger the animation.
+    setTimeout(() => {
+      this.setState({show: true});
+    }, 10);
+  }
+
+  removeOverlay() {
+    this.setState({signIn: true});
+    setTimeout(() => {
+      // Allow scrolling and fully remove overlay after animation finishes.
+      document.body.classList.remove("welcome");
+    }, 400);
+  }
+
+  onInputChange(e) {
+    this.setState({emailInput: e.target.value});
+  }
+
+  onSubmit() {
+    this.props.dispatch(ac.UserEvent({event: "SUBMIT_EMAIL"}));
+    window.addEventListener("focus", this.removeOverlay);
```

Comment:
can you use `visibilitychange` event instead of `focus` ? and also remove the `eventListener` once you've removed the overlay, or else this will fire every time you focus back to the page

Reply:
`focus` isn't really used much anymore since HTML5 uses the page visibility API. Plus `focus` could be fired if the browser loses focus other ways, but that doesn't mean it's not visible. `visibilitychange` feels like the more suited event